### ISSUE TITLE
Programmable renderer is broken

### ISSUE THREAD

**Issue by mruegenberg** [paulobarcelos/ofxWarpBlendTool#5]
When using the programmable renderer, nothing is displayed.

e.g the following `main.cpp` works:

```
int main( ){
    ofGLWindowSettings settings;
    settings.setGLVersion(2,1); // last version where fixed pipeline was default
    ofCreateWindow(settings);

    ofRunApp(new ofApp());
}
```

![scn1](https://cloud.githubusercontent.com/assets/78885/15066944/5accb984-136d-11e6-977f-7b4c3a1a01e0.png)

The following uses the programmable renderer and fails:

```
ofGLWindowSettings settings;
settings.setGLVersion(3,0); // first version with programmable renderer by default
ofCreateWindow(settings);
ofRunApp(new ofApp());
```

![scn2](https://cloud.githubusercontent.com/assets/78885/15066962/6f36f24a-136d-11e6-8a0a-a960b6e26b23.png)

Do you have any idea what might be causing this?

**Comment by mruegenberg** [comment]
One part of the problem might be the fixed function pipeline stuff in `Controller::draw()`, which can easily be replaced with drawing based on ofMesh.

It still doesn't work after making this change, though, so it might also be necessary to update the shaders for current OpenGL.

Unfortunately, ofGLWarper also seems to somewhat break when using the programmable renderer (see roymacdonald/ofxGLWarper#8 ; fixed by roymacdonald/ofxGLWarper#9).

**Comment by mruegenberg** [comment]
Ok, ofxGLWarper should work now.

I also added drawing based on ofMesh as well as shaders for the programmable renderer.
